python/cudaq/kernel/ast_bridge.py

```diff
@@ -2720,6 +2720,16 @@ def visit_Compare(self, node):
             return
 
         if isinstance(op, ast.NotEq):
+            if F64Type.isinstance(left.type) and IntegerType.isinstance(
+                    comparator.type):
+                left = arith.FPToSIOp(comparator.type, left).result
+            if IntegerType(left.type).width < IntegerType(
+                    comparator.type).width:
+                if IntegerType(left.type).width == 1:
+                    # For i1 (bool), always performs an unsigned extension.
+                    left = arith.ExtUIOp(comparator.type, left).result
+                else:
+                    left = arith.ExtSIOp(comparator.type, left).result
```

Comment:
Using `cc.CastOp` will simplify things potentially. It's one operation that can do sign-extension, zero-extension, truncation, from/to floating-point conversions, etc.

Reply:
Done in https://github.com/NVIDIA/cuda-quantum/pull/1654/commits/130ead60f4035e7cd1548164d71cb733434bb4b1
Agreed, this would also make the IR printing better, e.g., `cc.cast unsigned` is more readable than before.